### ISSUE TITLE
🐛 Quick fix not respecting linter being disabled

### ISSUE THREAD

**Issue by Akkuma** [biomejs/biome-vscode#292]
### VS Code version

1.91.1

### Extension version

2.3.0

### Biome version

1.8.3

### Operating system

- [ ] Windows
- [X] macOS
- [ ] Linux

### Description

biome's config file is not respected when it comes to lint quick fixes:
```
{
	"$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
	"vcs": {
		"enabled": true,
		"clientKind": "git",
		"defaultBranch": "main",
		"useIgnoreFile": true
	},
	"organizeImports": {
		"enabled": true
	},
	"formatter": {
		"enabled": true,
		"formatWithErrors": true,
		"indentWidth": 2,
		"lineWidth": 110,
		"ignore": ["package.json", ".ts", ".sst", "cdk.context.json"]
	},
	"linter": {
		"enabled": false
	},
	"javascript": {
		"formatter": {
			"quoteStyle": "single"
		}
	},
	"overrides": [
		{
			"include": ["extensions.json", "tsconfig.shared.json"],
			"json": { "parser": { "allowComments": true } }
		}
	]
}
```
<img width="602" alt="image" src="https://github.com/user-attachments/assets/01cc06fa-8841-4953-a102-285ccbdd5f7b">

My oxlint is detecting an issue and showing it to me, but biome is showing a quick fix that wouldn't even work as the linter is disabled.

### Steps to reproduce

1. Create biome with lint turned off
2. Use alternative linter
3. Create an overlapping rule that both have, in this case `no-empty-pattern`
4. Quick fix still gets populated despite

### Expected behavior

Biome checks whether the linter is even enabled before providing linting suggestions

### Does this issue occur when using the CLI directly?

No

### Logs

```plain text
[CodeAction(CodeAction { title: "Suppress rule lint/correctness/noEmptyPattern", kind: Some(CodeActionKind("quickfix.suppressRule.biome.correctness.noEmptyPattern")), diagnostics: None, edit: Some(WorkspaceEdit { changes: Some({Url { scheme: "file", cannot_be_a_base: false, username: "", password: None, host: None, port: None, path: "/Users/omni/git/payment-sst/packages/services/core/user/user-service.test.ts", query: None, fragment: None }: [TextEdit { range: Range { start: Position { line: 30, character: 1 }, end: Position { line: 30, character: 1 } }, new_text: "// biome-ignore lint/correctness/noEmptyPattern: <explanation>\n\t" }]}), document_changes: None, change_annotations: None }), command: None, is_preferred: None, disabled: None, data: None })]
┌─┘
└─┐biome_lsp::handlers::analysis::code_actions{uri=file:///Users/omni/git/payment-sst/packages/services/core/user/user-service.test.ts, range=Range { start: Position { line: 30, character: 13 }, end: Position { line: 30, character: 15 } }, only=Some([CodeActionKind("quickfix")]), diagnostics=[Diagnostic { range: Range { start: Position { line: 30, character: 13 }, end: Position { line: 30, character: 15 } }, severity: Some(Warning), code: None, code_description: None, source: Some("oxc"), message: "eslint(no-empty-pattern): Disallow empty destructuring patterns.\nhelp: Passing `null` or `undefined` will result in runtime error because `null` and `undefined` cannot be destructured.", related_information: Some([DiagnosticRelatedInformation { location: Location { uri: Url { scheme: "file", cannot_be_a_base: false, username: "", password: None, host: None, port: None, path: "/Users/omni/git/payment-sst/packages/services/core/user/user-service.test.ts", query: None, fragment: None }, range: Range { start: Position { line: 30, character: 13 }, end: Position { line: 30, character: 15 } } }, message: "Empty object binding pattern" }]), tags: None, data: None }]}
  ├─  0ms DEBUG biome_service::workspace::server File capabilities: Js(JsFileSource { language: TypeScript { definition_file: false }, variant: Standard, module_kind: Module, version: ES2022, embedding_kind: None }) BiomePath { path: "/Users/omni/git/payment-sst/packages/services/core/user/user-service.test.ts" }
  ├─  0ms DEBUG biome_service::workspace The file has the following feature sets: 
  │ {Format: Supported, Search: FileNotSupported, Lint: FeatureNotEnabled, OrganizeImports: Supported}
  ├─  0ms DEBUG biome_lsp::handlers::analysis Cursor range 943..945
  ├─  0ms DEBUG biome_service::workspace::server File capabilities: Js(JsFileSource { language: TypeScript { definition_file: false }, variant: Standard, module_kind: Module, version: ES2022, embedding_kind: None }) BiomePath { path: "/Users/omni/git/payment-sst/packages/services/core/user/user-service.test.ts" }
  └─┐biome_service::file_handlers::javascript::code_actions{}
    └─┐biome_service::file_handlers::javascript::Code actions JavaScript{range=943..945, path=BiomePath { path: "/Users/omni/git/payment-sst/packages/services/core/user/user-service.test.ts" }}
```
